File: api/app/schemas/llm.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LLMApiKeyResponse(BaseModel):
    id: str
    model_id: str
    name: str | None = None
    is_active: bool
    masked_key: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class LLMModelResponse(BaseModel):
    id: str
    provider: str
    model: str
    is_active: bool
    context_window: int | None = None
    api_keys: list[LLMApiKeyResponse]

    model_config = ConfigDict(from_attributes=True)


class AddApiKeyRequest(BaseModel):
    model_id: str
    name: str | None = None
    api_key: str


class AddApiKeyResponse(BaseModel):
    id: str
    model_id: str
    name: str | None = None
    is_active: bool
    masked_key: str
    created_at: datetime


class LLMSettingsResponse(BaseModel):
    default_llm_model_id: str | None = None
    default_llm_model: LLMModelResponse | None = None
    user_comm_model_id: str | None = None
    user_comm_model: LLMModelResponse | None = None

    model_config = ConfigDict(from_attributes=True)


class LLMSettingsUpdate(BaseModel):
    default_llm_model_id: str | None = None
    user_comm_model_id: str | None = None

File: web/src/types/llm.ts
export interface LLMApiKeyItem {
  id: string;
  model_id: string;
  name?: string | null;
  is_active: boolean;
  masked_key: string;
  created_at: string;
}

export interface LLMModelItem {
  id: string;
  provider: string;
  model: string;
  is_active: boolean;
  api_keys: LLMApiKeyItem[];
}

export interface LLMSettingsResponse {
  default_llm_model_id: string | null;
  default_llm_model: LLMModelItem | null;
  user_comm_model_id: string | null;
  user_comm_model: LLMModelItem | null;
}

export interface LLMSettingsUpdate {
  default_llm_model_id?: string | null;
  user_comm_model_id?: string | null;
}

export interface PromptTemplateResponse {
  pre_prompt: string;
  ending_prompt: string;
  chat_assistant: string;
  parse_query: string;
  summarize: string;
  title: string;
}

export interface PromptTemplateUpdate {
  pre_prompt?: string | null;
  ending_prompt?: string | null;
  chat_assistant?: string | null;
  parse_query?: string | null;
  summarize?: string | null;
  title?: string | null;
}

export interface SystemSettingItem {
  key: string;
  value: string;
}

export interface ScraperHeaderItem {
  id: string;
  name: string;
  header: string | null;
  status: "ready" | "error";
  error_message: string | null;
  error_at: string | null;
  updated_at: string;
}

export interface ScraperHeaderUpdate {
  header: string;
}

export interface EmbedModelItem {
  id: string;
  model_name: string;
  display_name: string;
  is_active: boolean;
}

export interface SystemSettingUpdate {
  key: string;
  value: string;
}

export interface EmbedTextRequest {
  text: string;
}

export interface EmbedTextResponse {
  model: string;
  dimensions: number;
  embedding: number[];
}
